src/zimcheck/checks.cpp

```diff
@@ -590,21 +591,21 @@ class TaskDispatcher
 {
 public: // functions
     explicit TaskDispatcher(ArticleChecker* ac)
-        : taskStream(ac)
+        : taskStream(new TaskStream(ac))
     {}
 
     void addTask(zim::Entry entry)
     {
-        taskStream.addTask(entry);
+        taskStream->addTask(entry);
     }
 
-    void noMoreTasks()
+    void waitForAllTasksToComplete()
     {
-        taskStream.noMoreTasks();
+        taskStream.reset();
```

Comment:
> It is a bit tricky to delete the pointer to have the destructor wait for the thread.

What's the problem with it? Such a destructor of `TaskStream` enables the following concise implementation of `TaskDispatcher::waitForAllTasksToComplete()` in a subsequent commit: 

```
    void waitForAllTasksToComplete()
    {
        taskStreams.clear();
    }
```

Reply:
I'm a bit disturbed because we are using a side effect of deleting the task to wait for all task.

We may call `waitForAllTasksToComplete` because we need to kind of sync point to reached and then continue to add task to the stream. But `waitForAllTasksToComplete` does more than wainting, it also deletes all the task streams and so makes the `TaskDispatcher` in unusable state.
If we really want to correctly close the taskStream by using it destructor, we should destruct the `taskStream` in the `TaskDispatcher` destructor.

Reply:
I tried to address your concern, but that turned out to be not that easy. I don't want to waste time on functionality that we don't currently need, so I simply renamed the `waitForAllTasksToComplete()` function to `finish()` and added a corresponding comment above it.

Reply:
`finish()` and a good name should be enough.